File: src/main/java/io/github/takenoha/towerdefense/persistence/VoucherWithdrawalResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;

/** Result of a wallet debit plus one durable voucher creation. */
public record VoucherWithdrawalResult(
        OperationOutcome outcome,
        ResourceVoucher voucher) {
    public VoucherWithdrawalResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(voucher, "voucher");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/VoucherWithdrawalResult.kt
package io.github.takenoha.towerdefense.persistence

import kotlin.jvm.JvmRecord

/** Result of a wallet debit plus one durable voucher creation. */
@JvmRecord
data class VoucherWithdrawalResult(
    val outcome: OperationOutcome,
    val voucher: ResourceVoucher,
)
